### PR TITLE
Improvements to notmuch module

### PR DESCRIPTION
**Pull request by mnick**
This PR adds various features to the notmuch module

- Filter read messages in notmuch show (by folding them automatically)
- Refresh all notmuch buffers after sync (via process sentinel)
- Support custom sync commands (introduces an additional defvar)
- Add command to select From email prior to composing via ivy

I wasn't sure if you prefer separate diff or one big diff. Please let me know if you want me to split this into separate PRs.

### CODE REVIEW

File: modules/email/notmuch/autoload.el [hunk 1]
```diff
@@ -68,6 +86,14 @@
   (notmuch-tree-add-tag (list "+spam" "-inbox" "-unread"))
   (notmuch-tree-next-message))
 
+;;;###autoload
+(defun +notmuch/ivy-compose ()
+  "Compose new mail"
+  (interactive)
+  (ivy-read "From: "
+            (notmuch-user-emails)
+            :action (lambda (selection) (notmuch-mua-mail nil nil (list (cons 'From selection))))))
```
Comment (hlissner) [inline modules/email/notmuch/autoload.el]: Could you generalize this to use `completing-read`? This way ivy, helm, ido or other users can use this command.

File: modules/email/notmuch/config.el [hunk 2]
```diff
@@ -42,6 +45,9 @@
 
   ;; (setq-hook! 'notmuch-show-mode-hook line-spacing 0)
 
+  ;; only unfold unread messages in thread by default
+  (add-hook 'notmuch-show-hook '+notmuch-show-expand-only-unread-h)
```
Comment (hlissner) [inline modules/email/notmuch/config.el]: Minor issue: please sharp-quote function symbols. i.e. `#'+notmuch-show-expand-only-unread-h`

Comment (mnick) [reply modules/email/notmuch/config.el]: Resolved in e279409

File: modules/email/notmuch/autoload.el (hunk 1 above)
Comment (mnick) [reply modules/email/notmuch/autoload.el]: Resolved in a4050a1

**Comment by mnick** [comment]
Thanks for the review. I also added some basic key bindings under localleader.

File: modules/email/notmuch/config.el [hunk 3]
```diff
@@ -56,7 +56,20 @@
...
+   :map (notmuch-search-mode-map notmuch-tree-mode-map notmuch-show-mode-map)
+   :desc "compose email" "c" #'+notmuch/compose
+   :desc "fetch new email" "u" #'+notmuch/update
+   :desc "quit notmuch" "q" #'+notmuch/quit
+   :map notmuch-search-mode-map
+   :desc "mark as deleted" "d" #'+notmuch/search-delete
+   :desc "mark as spam" "d" #'+notmuch/search-spam
+   :map notmuch-tree-mode-map
+   :desc "mark as deleted" "d" #'+notmuch/tree-delete
+   :desc "mark as spam" "d" #'+notmuch/tree-spam))
```
Comment (hlissner) [inline modules/email/notmuch/config.el]: It appears the `d` key is bound twice in each map. Was the second `d` meant to be `D`? Or a different key?

Comment (mnick) [reply modules/email/notmuch/config.el]: Good catch, thanks!

**Comment by hlissner** [comment]
Looks good to me! Thanks for the help!

**Pull request merged**